src/app/view/directives/dot-maxlength/dot-maxlength.directive.ts

```diff
@@ -7,29 +9,78 @@ import { Directive, ElementRef, HostListener, Input } from '@angular/core';
 @Directive({
     selector: '[dotMaxlength]'
 })
-export class DotMaxlengthDirective {
+export class DotMaxlengthDirective implements OnInit, OnDestroy {
     private _maxLength: number;
+    private events = ['paste', 'keypress'];
+    private destroy$: Subject<boolean> = new Subject<boolean>();
 
     constructor(private el: ElementRef) {}
 
+    ngOnInit() {
+        const eventStreams = this.events.map(ev => fromEvent(this.el.nativeElement, ev));
+        const allEvents$ = merge(...eventStreams);
```

Comment:
Why the `...` here?

Reply:
a way to do it, `export function merge<T>(...observables: (ObservableInput<T> | SchedulerLike | number)[]): Observable<T>;`